src/main/java/com/jcabi/jdbc/StoredProcedureOutcome.java

```diff
@@ -0,0 +1,110 @@
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.jdbc;
+
+import com.jcabi.aspects.Immutable;
+import java.sql.CallableStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+/**
+ * Outcome of a stored procedure with OUT parameters.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.17
+ * @param <T> Type of the returned result, which <b>has to be</b> Object[]
+ */
+@Immutable
+@ToString
+@EqualsAndHashCode(of = "indexes")
+public final class StoredProcedureOutcome<T> implements Outcome<T> {
+
+    /**
+     * OUT parameters' indexes.
+     */
+    @Immutable.Array
+    private final transient int[] indexes;
+
+    /**
+     * Ctor.
+     * @param params Number of OUT params. Has to be > 0.
+     *  If this ctor is used, it is assumed that the OUT parameters
+     *  are from index 1 to including nrop.
+     */
+    public StoredProcedureOutcome(final int params) {
+        if (params <= 0) {
+            throw new IllegalArgumentException(
+                "Nr of out params has to be a positive int!"
+            );
+        }
+        this.indexes = new int[params];
+        for (int idx = 0; idx < params; ++idx) {
+            this.indexes[idx] = idx + 1;
+        }
+    }
+
+    /**
+     * Ctor.
+     * @param indexes Indexes of the OUT params.
+     *  <b>Index count starts from 1</b>.
+     */
+    public StoredProcedureOutcome(final int... indexes) {
+        if (indexes.length == 0) {
+            throw new IllegalArgumentException(
+                "At least 1 OUT param's index needs to be specified!"
+            );
+        }
+        final int size = indexes.length;
+        this.indexes = new int[size];
+        for (int idx = 0; idx < size; ++idx) {
```

Comment:
@mkordas I see... it really is, and it cannot be removed. Usually we call one ctor from another one using ``this(...)`` to avoid duplication, but here it is not possible since first we have to turn the ``int`` from the first ctor, in an array, and the call ``this(...)`` has to be the first. 

I removed the first ctor and updated the PR description accordingly. It was in fact a convenience ctor and I figured it's not worth the duplication.